pkg/daemon/metrics.go

```diff
@@ -205,8 +211,9 @@ func extractMetrics(configName, processName string, ifaces []string, output stri
 					r := []rune(ifaces[portId-1])
 					masterOffsetIfaceName[configName] = string(r[:len(r)-1]) + "x"
 					slaveIfaceName[configName] = ifaces[portId-1]
-				} else if role == FAULTY {
-					if isSlaveFaulty(configName, ifaces[portId-1]) {
+				} else if role == FAULTY { // only if ptp4l is processing offset, ts2phc offset won't effect by port faulty
+					if isSlaveFaulty(configName, ifaces[portId-1]) &&
+						getMasterSourceProcess(configName) == ptp4lProcessName {
```

Comment:
https://github.com/openshift/linuxptp-daemon/pull/116/commits/fa600556888fcdab8409c402f9477fe2b5559817#diff-1ba451acf68704ac9de95e07f9b218be2fcb2bc954a8398309bbbf32535e3251R206
Would this ever not be ptp4l?